src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -701,16 +701,46 @@ private static function recursively_iterate_json( $dir ) {
 		return $nested_json_files;
 	}
 
+	/**
+	 * Determines if a supplied style variation matches the provided scope.
+	 *
+	 * For backwards compatibility, if a variation does not define any scope
+	 * related property, e.g. `blockTypes`, it is assumed to be a theme style
+	 * variation.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @param array  $variation Theme.json shaped style variation object.
+	 * @param string $scope     Scope to check e.g. theme, block etc.
+	 *
+	 * @return boolean
+	 */
+	private static function style_variation_has_scope( $variation, $scope ) {
+		if ( 'block' === $scope ) {
+				return isset( $variation['blockTypes'] );
+		}
+
+		if ( 'theme' === $scope ) {
+				return ! isset( $variation['blockTypes'] );
+		}
```

Comment:
```suggestion
		if ( 'block' === $scope ) {
			return isset( $variation['blockTypes'] );
		}

		if ( 'theme' === $scope ) {
			return ! isset( $variation['blockTypes'] );
		}
```

Reply:
Not sure what's going on there. I'll try and get it cleaned up.